src/FunctionApps/python/IntakePipeline/conversion.py

```diff
@@ -1,18 +1,62 @@
 import logging
 import re
 import requests
+import hl7
 from typing import List, Dict
 
 
-def clean_message(message: str, delimiter: str = "\n") -> str:
-    cleaned_message = re.sub("[\r\n]+", delimiter, message)
+def clean_message(message: str) -> str:
+    parsed_message = None
+    try:
+        parsed_message = hl7.parse(message)
+
+        # MSH-7
+        cleansed_datetime = normalize_hl7_datetime(parsed_message[0][7][0])
+        parsed_message[0][7][0] = cleansed_datetime
+
+    except Exception:
+        logging.exception(
+            "Exception occurred while cleaning message.  "
+            + "Passing through original message."
+        )
+
+        return message
+
+    return str(parsed_message)
+
+
+def normalize_hl7_datetime(hl7_datetime: str) -> str:
+    hl7_datetime_parts = re.match(
+        r"(\d+)(\.)?(\d+)?([+-])?(\d+)?", hl7_datetime
+    ).groups()
+
+    # Date Base
+    normalized_datetime = hl7_datetime_parts[0][:14]  # First 14 digits
+
+    # Date Decimal
+    if hl7_datetime_parts[2]:
+        normalized_datetime += (
+            hl7_datetime_parts[1] + hl7_datetime_parts[2][:4]
+        )  # . plus first 4 digits
+
+    # Date Timezone
+    if hl7_datetime_parts[4] and len(hl7_datetime_parts[4]) >= 4:
+        normalized_datetime += (
+            hl7_datetime_parts[3] + hl7_datetime_parts[4][:4]
+        )  # +/- plus 4 digits
```

Comment:
Same comment as above. Would a built-in library be suitable to more robustly handle these formatting conversions?